human_aware_rl/rllib/utils.py

```diff
@@ -0,0 +1,41 @@
+from overcooked_ai_py.mdp.overcooked_mdp import OvercookedGridworld
+from overcooked_ai_py.mdp.overcooked_env import OvercookedEnv
+from overcooked_ai_py.planning.planners import MediumLevelPlanner, NO_COUNTERS_PARAMS
+import numpy as np
+import inspect
+
+def softmax(logits):
+    e_x = np.exp(logits.T - np.max(logits))
+    return (e_x / np.sum(e_x, axis=0)).T
+
+def get_base_env(mdp_params, env_params):
+    mdp_fn = lambda: OvercookedGridworld.from_layout_name(**mdp_params)
+    base_env = OvercookedEnv(mdp_fn, **env_params)
+    return base_env
+
+def get_mlp(env):
+    return MediumLevelPlanner.from_pickle_or_compute(env.mdp, NO_COUNTERS_PARAMS, force_compute=False)
```

Comment:
OK I ended up using `AgentEvaluator` as a container for all 3 as you suggested, but my one concern is that `AgentEvaluator` creates the environment itself, under the hood, rather than being environment agnostic. I think this might pose a problem later, if the `OvercookedMultiAgent` environment wrapper ever introduces functionality that `AgentEvaluator` needs, or if, for example, we decide to introduce additional environments (i.e. a toy environment)

Reply:
Yeah that's a good point. One thing to keep in mind is that most likely we will also be running experiments on a competitive environment at some point, so even if we don't use a toy environment, you should be expecting at some point to have to add multi-environment support to the various aspects of the code. When I first coded up the competitive environment `gathering_ai` (that is in my `human_model_theory` branch), I was under deadline pressure and didn't have time to think properly about how to do this or refactor the old code, so I just duplicated a bunch of files and added hacky flags everywhere. But given that we're currently with less pressure, we should probably do this nicely.